### ISSUE TITLE
Unable to apply muiTheme with QueryBuilder

### ISSUE THREAD

**Issue by humoyun91** [react-querybuilder/react-querybuilder#321]
First of all, I would like to thank for this amazing library. We have been struggling with our dashboard in which there was a native filter input so that our users can provide mongo-db queries when fetching resources. But its usability was a pain in the ass until we found this library to build nice DSL with forms. Now users can just ask anything without knowing the mongo db query syntax. 
However, as our frontend theme was only dark, the some default text colors (especially Select) were barely visible so we tried to change it through `muiTheme` which will be passed to wrapper ThemeProvider for `QueryBuilder`. To our surprise nothing seems to work, can you please help or at least show guidance how to achieve our goal. Thanks in advance!

> I checked `query-builder` source code, I am not sure but it seems to me there is no way to customize `muiTheme` from outside. Maybe I missed something!
> Also I installed `query-builder` in a new clean react project but got same result!

Here is relevant code snippets from our codebase:
```
// just tried every possible way, here is just some samples
const muiTheme = createTheme({
  palette: {
    primary: {
      main: "#fff",
      light: "#fff",
    },
    text: {
      primary: "#fff",
      secondary: "#fff",
      disabled: "#fff",
    },
  },
  components: {
    MuiInput: {
      styleOverrides: {
        root: {
          color: "#fff",
        },
      },
    },
    MuiFormControl: {
      styleOverrides: {
        root: {
          color: "#fff",
        },
      },
    },

    MuiSelect: {
      styleOverrides: {
        standard: {
          color: "#fff",
        },
      },
    },
  },
});
```

```
<Container>
  <Typography>{filterName} filter</Typography>
  <ThemeProvider theme={muiTheme}>
    <QueryBuilder
      disabled={queryDisabled}
      fields={fields}
      query={query}
      operators={operators}
      combinators={combinators}
      onQueryChange={handleQueryChange}
      controlElements={materialControlElements}
    />
  </ThemeProvider>
</Container>
```
### Here is the screenshot of our dashboard:

![ss](https://user-images.githubusercontent.com/85469622/172508992-a5287063-621e-451e-9497-5f64c43038cb.png)

**Comment by jakeboone02** [comment]
I can look at this closer tomorrow (I'm in the western US), but in the meantime could you set up a codesandbox to demonstrate the issue? You can use [this starter template](https://codesandbox.io/s/github/react-querybuilder/react-querybuilder/tree/main/examples/material) that uses the MUI components.

**Comment by jakeboone02** [comment]
Honestly I'm struggling with this a bit. I opened #324 hoping it would be fairly straightforward. Seemed like I could just add a `useTheme()` hook to get the theme config from context, wrap the components in their own `<ThemeProvider />` using the aforementioned hook's return value, and call it a day. But it simply doesn't work.

Strangely, it _does_ work when I run the demo locally. The only difference appears to be that the demo imports the components directly from their `src` directory within the monorepo structure. I tried publishing the src folder as part of the `@react-querybuilder/material` package but couldn't get that working either.

Any help or direction you might be able to provide would be greatly appreciated.

FWIW, there is a way to override the native props of the style library for all the compatibility component packages. See #325 for an example using the MUI package.

**Comment by jakeboone02** [comment]
Trying something: an alternate version of [this codesandbox](https://codesandbox.io/s/nifty-murdock-xhqhcq?file=/src/App.tsx) should appear in #324 automatically using the updated packages.

**Comment by jakeboone02** [comment]
Good news, everyone! I finally cracked the case of the missing theme configurations (see #324) and the issue will be resolved in v5.0, due out soon.

**Comment by GabriFila** [comment]
> Trying something: an alternate version of [this codesandbox](https://codesandbox.io/s/nifty-murdock-xhqhcq?file=/src/App.tsx) should appear in #324 automatically using the updated packages.


Hi, great library!
I have some problems with passing my mui theme values to the querybuilder, I looked at the [codesandbox](https://codesandbox.io/s/nifty-murdock-xhqhcq?file=/src/App.tsx) you cited but it doesn't seem to work, it throws an error. I used the `5.0.0-alpha.3` for both the main package and the material adapter.

Do you have any suggestions/ideas on how to solve it? I might be using the wrong configuration.

Thank you!

**Comment by jakeboone02** [comment]
Hi @GabriFila. That sandbox is out of date now. [This one](https://codesandbox.io/s/busy-khayyam-hq236o?file=/src/App.tsx) has updated code, but there's still an issue with the dynamic component imports that crashes the page. I'm working on it.

**Comment by GabriFila** [comment]
Thank you, I'll wait on updates!

**Comment by jakeboone02** [comment]
Reopening this as it still seems to be broken in the latest v5 alpha.

**Comment by jakeboone02** [comment]
@humoyun91 @GabriFila @zzoman - I _think_ I've finally resolved the issues with `@react-querybuilder/material`. Can you all try out the "material example" CodeSandbox link in PR #381? ([This](https://codesandbox.io/s/react-querybuilder-material-example-vi6070?file=/src/App.tsx) is the current link.)

Note: if you try it as-is, there will be a slight delay before the MUI components render in place of the default components. The new docs [explain how to avoid this](https://react-querybuilder.js.org/docs/next/compat#preload-mui-components) flash of non-MUI.

**Comment by humoyun91** [comment]
@jakeboone02 

I tried to run it with commit hash based version which I took from the provided `codesandbox`:

![Screen Shot 2022-10-07 at 11 37 21 AM](https://user-images.githubusercontent.com/85469622/194456450-8fa9956d-0c54-4a2b-b477-0722291e33c7.png)

After this I am having issue with importing some types
```
import { Field, NameLabelPair } from 'react-querybuilder'
```

![Screen Shot 2022-10-07 at 11 39 54 AM](https://user-images.githubusercontent.com/85469622/194456600-f1ccbc16-7676-46cf-b0c8-8f44cbfd871f.png)

and seems this is also the case for codesandbox itself (though it is showing correct mui applied)

![Screen Shot 2022-10-07 at 11 42 51 AM](https://user-images.githubusercontent.com/85469622/194456969-2a68e609-aace-4231-8b6c-1320b5e3bb43.png)

 Because of this, our codebase is failing to compile and I cannot assert whether mui is applied correctly

**Comment by jakeboone02** [comment]
Thanks, @humoyun91. I'll see if I can get that resolved.

**Comment by jakeboone02** [comment]
@humoyun91 that should be resolved now.

New MUI example link is https://codesandbox.io/s/react-querybuilder-material-example-93jfoh

**Comment by jakeboone02** [comment]
I went ahead and released that PR as `5.0.0-alpha.8` so you can try it out as a proper release instead of getting the commit-hashed version from CodeSandbox.

If it actually fixes this issue, then it'll be more like a release candidate than an alpha since there's [not much left on the 5.0 milestone](https://github.com/react-querybuilder/react-querybuilder/milestone/2). I won't be able to release 5.0 proper until mid-to-late next week in any case.
